apigw-sqs-lambda-ddb/README.md

```diff
@@ -0,0 +1,75 @@
+* [Git Installed](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)
+* [AWS Serverless Application Model](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/serverless-sam-cli-install.html) (AWS SAM) installed
+
+## Deployment Instructions
+
+1. Create a new directory, navigate to that directory in a terminal and clone the GitHub repository:
+    ```
+    git clone https://github.com/aws-samples/serverless-patterns
+    ```
+1. Change directory to the pattern directory:
+    ```
+    cd apigw-sqs-lambda-ddb
+    ```
+1. From the command line, use AWS SAM to deploy the AWS resources for the pattern as specified in the template.yml file:
+    ```
+    sam deploy --guided
+    ```
+1. During the prompts:
+    * Enter a stack name
+    * Enter the desired AWS Region
+    * Allow SAM CLI to create IAM roles with the required permissions.
+
+    Once you have run `sam deploy -guided` mode once and saved arguments to a configuration file (samconfig.toml), you can use `sam deploy` in future to use these defaults.
+
+1. Note the outputs from the SAM deployment process. These contain the resource names and/or ARNs which are used for testing.
+
+## How it works
+
+<img src="docs/apigw-sqs-lambda-ddb.drawio.png" alt="architecture diagram"/>
+
+- This pattern deploys an Amazon API Gateway HTTP API with route /submit configured with basic authentication.
+- On receiving a request, API Gateway will invoke a Lambda authorizer which validates the request and returns a policy informing API Gateway to accept or deny the request.
+- When request is accepted, API Gateway sends the message payload to a queue in SQS. SQS uses another queue as Dead Letter Queue to send the messages in case of continued failures to process the messages from downstream lambda.
+- Messages from SQS is posted to a lambda function to process them.
+- Lambda function receives the messages from SQS and saves them into a DynamoDB table.
+
+## Testing
+
+Once the application is deployed:
+- Retrieve the HttpApiEndpoint value from CloudFormation Outputs
+- Retrieve the username and password from Secrets Manager in AWS Console.
+- Invoke the endpoint from Postman using some json payload and verify the payload saved in DynamoDB.
+
+Example POST Request: https://{HttpApiEndpoint}.execute-api.us-east-1.amazonaws.com/submit
+- Request Header: "Content-Type: application/json"
+- Request Header: "Authorization: Basic <credentials>" (where credentials is the Base64 encoding of ID and password joined by a single colon :)
+- Request Body: {"eventId":"value1", "message":"event message for testing"}
+
+## Cleanup
+
+1. Delete the stack
+    ```bash
+    aws cloudformation delete-stack --stack-name STACK_NAME
+    ```
+1. Confirm the stack has been deleted
+    ```bash
+    aws cloudformation list-stacks --query "StackSummaries[?contains(StackName,'STACK_NAME')].StackStatus"
+    ```
+
+This pattern was contributed by Ravi Kiran Ganji.
```

Comment:
this can be removed as we do the attribution differently

Reply:
done